### PR TITLE
Homebrew Security Audit

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Add the long delayed security audit post with timeline details.

### CODE REVIEW

File: _posts/2022-05-15-homebrew-security-audit.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: Security Audit
+author: MikeMcQuaid
+---
+
+Homebrew has had a paid security audit and addressed all flagged issues. This blog post has been a long time coming, apologies for the delay.
+
+Here's an overview of the timescale:
+
+- 11th June 2020: the [Mozilla Open Source program (MOSS)](https://www.mozilla.org/en-US/moss/) reaches out to Homebrew as we were nominated for a paid, sponsored security audit by [Radically Open Security (ROS)](https://radicallyopensecurity.com/)
```
Comment (samford) [inline _posts/2022-05-15-homebrew-security-audit.md:10]:
```suggestion
- 11th June 2020: The [Mozilla Open Source program (MOSS)](https://www.mozilla.org/en-US/moss/) reaches out to Homebrew as we were nominated for a paid, sponsored security audit by [Radically Open Security (ROS)](https://radicallyopensecurity.com/)
```

File: _posts/2022-05-15-homebrew-security-audit.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- 11th June 2020: Homebrew is introduced to ROS and provides the main areas of focus:
+  - macOS sandbox escapes
+  - CI/development workflow issues (e.g. ways to exploit our CI infrastructure or deploy changes that haven’t been reviewed)
+  - bad uses/setting/checking of Unix permissions
```
Comment (samford) [inline _posts/2022-05-15-homebrew-security-audit.md:14]:
```suggestion
  - Bad uses/setting/checking of Unix permissions
```

File: _posts/2022-05-15-homebrew-security-audit.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  - formulae being able to modify the [Homebrew/brew](https://github.com/Homebrew/brew) source process
```
Comment (samford) [inline _posts/2022-05-15-homebrew-security-audit.md:15]:
```suggestion
  - Formulae being able to modify the [Homebrew/brew](https://github.com/Homebrew/brew) source process
```

File: _posts/2022-05-15-homebrew-security-audit.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+- 18th June 2020: Homebrew meets with ROS to further discuss the audit, scope and process and provide access to ROS systems (e.g. GitLab, RocketChat)
+- 23rd September 2020: MOSS and ROS confirm contract
+- 14th October 2020: ROS begins security audit
+- October 2020 - March 2021: ROS communicates issues to Homebrew which are resolved e.g. with <https://github.com/Homebrew/brew/pull/10970> and <https://github.com/Homebrew/brew/pull/10972>
```
Comment (samford) [inline _posts/2022-05-15-homebrew-security-audit.md:19]:
```suggestion
- October 2020 - March 2021: ROS communicates issues to Homebrew which are resolved, e.g. with <https://github.com/Homebrew/brew/pull/10970> and <https://github.com/Homebrew/brew/pull/10972>
```

File: _posts/2022-05-15-homebrew-security-audit.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- 31st March 2021: ROS provides final security audit report to Homebrew
+- 4th April 2021: Homebrew provides a related [security incident disclosure](https://brew.sh/2021/04/21/security-incident-disclosure/) based on follow-up work.
```
Comment (samford) [inline _posts/2022-05-15-homebrew-security-audit.md:21]:
```suggestion
- 4th April 2021: Homebrew provides a related [security incident disclosure](https://brew.sh/2021/04/21/security-incident-disclosure/) based on follow-up work
```

The other bullet points don't end with a period, so this is simply for the sake of consistency.

**Review by samford — COMMENTED** [review]
From a stylistic point of view, there's maybe something to be said for standardizing the first word after the date colon and the first word of the bullet points (with exceptions like "macOS", of course). It looks more proper to me in this context but feel free to pass on this.

**Review by jacobbednarz — APPROVED** [review]

**Review by SMillerDev — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

File: _posts/2022-05-15-homebrew-security-audit.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+---
+title: Security Audit
+author: MikeMcQuaid
+---
+
+Homebrew has had a paid security audit and addressed all flagged issues. This blog post has been a long time coming, apologies for the delay.
```
Comment (EricFromCanada) [inline _posts/2022-05-15-homebrew-security-audit.md:6]: Either `coming;` or `coming, so`

File: _posts/2022-05-15-homebrew-security-audit.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+Here's an overview of the timescale:
+
+- 11th June 2020: the [Mozilla Open Source program (MOSS)](https://www.mozilla.org/en-US/moss/) reaches out to Homebrew as we were nominated for a paid, sponsored security audit by [Radically Open Security (ROS)](https://radicallyopensecurity.com/)
+- 11th June 2020: Homebrew is introduced to ROS and provides the main areas of focus:
```
Comment (EricFromCanada) [inline _posts/2022-05-15-homebrew-security-audit.md:11]: Subject/object confusion here. If Homebrew provided the main areas of focus, it should be the subject, e.g. `Homebrew meets with ROS and…`

File: _posts/2022-05-15-homebrew-security-audit.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  - macOS sandbox escapes
+  - CI/development workflow issues (e.g. ways to exploit our CI infrastructure or deploy changes that haven’t been reviewed)
+  - bad uses/setting/checking of Unix permissions
+  - formulae being able to modify the [Homebrew/brew](https://github.com/Homebrew/brew) source process
+- 18th June 2020: Homebrew meets with ROS to further discuss the audit, scope and process and provide access to ROS systems (e.g. GitLab, RocketChat)
```
Comment (EricFromCanada) [inline _posts/2022-05-15-homebrew-security-audit.md:16]: `ROS meets with Homebrew to…`

**Review by EricFromCanada — APPROVED** [review]

File: _posts/2022-05-15-homebrew-security-audit.md (hunk 6 above)
Comment (MikeMcQuaid) [inline _posts/2022-05-15-homebrew-security-audit.md:6]:
```suggestion
Homebrew has had a paid security audit and addressed all flagged issues. This blog post has been a long time coming; apologies for the delay.
```

File: _posts/2022-05-15-homebrew-security-audit.md (hunk 7 above)
Comment (MikeMcQuaid) [inline _posts/2022-05-15-homebrew-security-audit.md:11]:
```suggestion
- 11th June 2020: Homebrew meets with ROS and provides the main areas of focus:
```

File: _posts/2022-05-15-homebrew-security-audit.md (hunk 8 above)
Comment (MikeMcQuaid) [inline _posts/2022-05-15-homebrew-security-audit.md:16]:
```suggestion
- 18th June 2020: ROS meets with Homebrew to further discuss the audit, scope and process and provide access to ROS systems (e.g. GitLab, RocketChat)
```

**Comment by MikeMcQuaid** [comment]
@samford @ericfromcanada thanks for the help with wordsmithing!

**Comment by MikeMcQuaid** [comment]
@samford @ericfromcanada thanks for the help with wordsmithing!

**Pull request merged**